### PR TITLE
MMCA 4448 - Remove unused code and ToDos - customs-duty-deferment-frontend

### PR DESCRIPTION
**Pull request by HariHmrc**

### CODE REVIEW

File: app/controllers/ContactDetailsEditStartController.scala [hunk 1]
```diff
@@ -36,16 +36,16 @@ import scala.concurrent.{ExecutionContext, Future}
...
+                                                   dateTimeService: DateTimeService,
+                                                   identifier: IdentifierAction,
+                                                   resolveSessionId: SessionIdAction,
+                                                   accountLinkCacheService: AccountLinkCacheService,
+                                                   errorHandler: ErrorHandler,
+                                                   countriesProviderService: CountriesProviderService,
+                                                   appConfig: AppConfig,
+                                                   userAnswersCache: UserAnswersCache,
+                                                   mcc: MessagesControllerComponents
                                                  )(implicit ec: ExecutionContext)
```
Comment (AdamCoops) [inline app/controllers/ContactDetailsEditStartController.scala:49]: )(implicit ec: ExecutionContext) can probably be moved up a Line, just check it doesn't exceed char length

File: app/controllers/ContactDetailsEditStartController.scala [hunk 2]
```diff
@@ -82,7 +82,8 @@ class ContactDetailsEditStartController @Inject()(
         }
   }
 
-  private def setUserAnswers(initialContactDetails: ContactDetails, dutyDefermentDetails: DutyDefermentAccountLink, internalId: String, contactDetailsChange: Boolean): Option[UserAnswers] = {
+  private def setUserAnswers(initialContactDetails: ContactDetails, dutyDefermentDetails: DutyDefermentAccountLink,
```
Comment (AdamCoops) [inline app/controllers/ContactDetailsEditStartController.scala:85]: lets split these params out on new lines rather just some of them

File: app/controllers/EditAddressDetailsController.scala [hunk 3]
```diff
@@ -95,7 +96,7 @@ class EditAddressDetailsController @Inject()(view: edit_address_details,
   }
 
   private def updateContactDetailsUserAnswers(editAddressAnswers: EditAddressDetailsUserAnswers, eori: String)
```
Comment (AdamCoops) [inline app/controllers/EditAddressDetailsController.scala:98]: I think we want a new line for each param

File: app/controllers/EditContactDetailsController.scala [hunk 4]
```diff
@@ -95,7 +96,7 @@ class EditContactDetailsController @Inject()(view: edit_contact_details,
   }
 
   private def updateContactDetailsUserAnswers(editContactDetailAnswers: EditContactDetailsUserAnswers, eori: String)
```
Comment (AdamCoops) [inline app/controllers/EditContactDetailsController.scala:98]: new lines for the params as well here

File: app/mappings/EditContactDetailsFormProvider.scala [hunk 5]
```diff
@@ -22,11 +22,8 @@ import models.EditContactDetailsUserAnswers
...
 import play.api.data.{Form, FormError, Forms}
-import services.CountriesProviderService
 
-class EditContactDetailsFormProvider @Inject()(
-                                                countriesProviderService: CountriesProviderService
-                                              ) extends Constraints {
+class EditContactDetailsFormProvider @Inject()() extends Constraints {
 
   def apply(): Form[EditContactDetailsUserAnswers] = {
     // A hidden form component tracks countryName and is updated using JS.
```
Comment (AdamCoops) [inline app/mappings/EditContactDetailsFormProvider.scala:29]: this comment can be deleted

File: app/models/AccountLink.scala [hunk 6]
```diff
@@ -19,9 +19,9 @@ package models
 import play.api.libs.json.{Json, OFormat}
 
 case class AccountLink(eori: String, accountNumber: String,
```
Comment (AdamCoops) [inline app/models/AccountLink.scala:21]: I think these parms can be spaced out like the previously comments of this type

File: build.sbt [hunk 7]
```diff
@@ -14,21 +19,34 @@ lazy val microservice = Project(appName, file("."))
...
-    pipelineStages in Assets := Seq(gzip),
-    // ***************
-    // Use the silencer plugin to suppress warnings
-    scalacOptions += "-P:silencer:pathFilters=routes",
+    Assets / pipelineStages := Seq(gzip),
     libraryDependencies ++= Seq(
       compilerPlugin("com.github.ghik" % "silencer-plugin" % silencerVersion cross CrossVersion.full),
       "com.github.ghik" % "silencer-lib" % silencerVersion % Provided cross CrossVersion.full
     )
     // ***************
```
Comment (AdamCoops) [inline build.sbt:48]: delete this comment

File: project/plugins.sbt [hunk 8]
```diff
@@ -2,10 +2,11 @@ resolvers += MavenRepository("HMRC-open-artefacts-maven2", "https://open.artefac
...
 
-addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.15.0")
+addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.20.0")
 addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "2.2.0")
 addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.20")
 addSbtPlugin("com.typesafe.sbt" % "sbt-gzip" % "1.0.2")
 addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.9.3")
 addSbtPlugin("uk.gov.hmrc" % "sbt-play-cross-compilation" % "2.3.0")
 addSbtPlugin("io.github.irundaia" % "sbt-sassify" % "1.5.2")
+addSbtPlugin("org.scalastyle" %% "scalastyle-sbt-plugin" % "1.0.0")
```
Comment (AdamCoops) [inline project/plugins.sbt:12]: can we add an empty line before the file end here?

Comment (HariHmrc) [reply project/plugins.sbt:12]: end of file already have a empty line, could you review it again pls

**Comment by AdamCoops** [comment]
Can you fix this warning please?

ConfirmContactDetailsControllerSpec.scala:30:27: Unused import
[warn] import util.TestImplicits.RemoveCsrf

**Comment by AdamCoops** [comment]
And this one as well

ConfirmContactDetailsControllerSpec.scala:33:25: Unused import
[warn] import scala.concurrent.Future

**Comment by AdamCoops** [comment]
ShowContactDetailsControllerSpec.scala:30:25: Unused import
[warn] import uk.gov.hmrc.http.HttpClient

**Comment by AdamCoops** [comment]
MetadataSpec.scala:20:28: Unused import
[warn] import play.api.libs.json.{JsString, JsSuccess, Json, Reads, Writes}
[warn]                                              ^

                                                                                    ^

**Comment by AdamCoops** [comment]
MetadataSpec.scala:20:55: Unused import
[warn] import play.api.libs.json.{JsString, JsSuccess, Json, Reads, Writes}
[warn]                                                                                            ^

**Comment by AdamCoops** [comment]
MetadataSpec.scala:20:62: Unused import
[warn] import play.api.libs.json.{JsString, JsSuccess, Json, Reads, Writes}
[warn]                                                                                                           ^

**Comment by AdamCoops** [comment]
UserAnswersSpec.scala:24:20: Unused import
[warn] import scala.util.{Failure,

**Comment by AdamCoops** [comment]
frontend/test/utils/DateConvertersSpec.scala:35:33: Unused import
[warn] import models.{DDStatementType, FileRole}

**Comment by AdamCoops** [comment]
DateConvertersSpec.scala:38:19: Unused import
[warn] import viewmodels.DutyDefermentStatementPeriod

**Comment by AdamCoops** [comment]
SpecBase.scala:43:46: object JavaConverters in package collection is deprecated (since 2.13.0): Use `scala.jdk.CollectionConverters` instead
[warn]       val results = document.getElementsByTag("a").asScala.toList

**Comment by AdamCoops** [comment]
SpecBase.scala:58:46: object JavaConverters in package collection is deprecated (since 2.13.0): Use `scala.jdk.CollectionConverters` instead
[warn]       assert(!document.getElementsByAttribute("id").asScala.toList.exists(_.id() == id))

**Comment by AdamCoops** [comment]
Please delete the comments in ConfirmContactDetailsControllerSpec starting from line 38

**Comment by AdamCoops** [comment]
Please delete lines at 45 & 46 in ContactDetailsCache we only need one end of file line

**Comment by AdamCoops** [comment]
reformat the params for ContactDetailsCache

**Comment by AdamCoops** [comment]
reformat params and delete line 44 in UserAnswersCache

**Comment by AdamCoops** [comment]
Add newline after lines 41 and 43 in AppConfig
and delete line 111

**Comment by AdamCoops** [comment]
ErrorHandler has several lines that over the character length

**Comment by AdamCoops** [comment]
CustomsFinancialsApiConnector has several lines over the character length

**Comment by AdamCoops** [comment]
CDSAccountStatusId has unused comments in it

EditContactDetailsControllerSpec has commented out code; lets delete it

NotSubscribedControllerSpec; add space after line 24

ServiceUnavailableControllerSpec; add space after line 27 and line 23 is too long  address line length issues in ConstraintsSpec (180, 184, 195, 199, 213, 217, 241, and 242)

Make use of emptyString on line 26 of Utils, line 49 of ContactDetailsViewModel and various places in the specs.   Add a “ “ and a "-" to the utils folder and make use of them in the code as well. Write tests for them as well. There are examples of these in sides-stub if you want examples

**Comment by AdamCoops** [comment]
ContactDetailsViewModel; address line length issues on  37 and 39  Delete comment on 37 of DutyDefermentStatementPeriod and address line length issues on 56 and 58

Address line length issues in DutyDefermentStatementsForEori on 33 and 44.  Address line length issue on 43 of AccountLinkCacheServiceSpec

Address line length issue on 38 and 65 of AuditingServiceSpec

ContactDetailsCacheServiceSpec; line length on 64  DateTimeServiceSpec; remove excess empty lines at end of file. We only need one

FakeDataRetrievalAction; line length on 29

FakeIdentifierAction; line length on 29

TestData; line length on 41

DateConvertersSpec; remove excess empty lines at end of file. We only need one

**Comment by AdamCoops** [comment]
84, 92, 101 - line lengths for Constraints

EditAddressDetailsFormProvider - Remove comments at 34/35/62 and fix line lengths 40-42, 63, 66. Delete lines 73 and 74. We only need one space at the file end.

EditContactDetailsFormProvider; Delete lines 47 and 48 - removing the space and the comment. Fix line lengths of 49 and 52

**Review by AdamCoops — APPROVED** [review]

**Pull request merged**